org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/dtd/contentmodel/CMDTDDocument.java

```diff
@@ -200,6 +200,9 @@ private static int getEntityNameStartColumnNumber(String entityName, ScannedEnti
 			char[] ch = scannedEntity.ch;
 			int wordIndex = entityName.length(); //
 			int startEntityNameIndex = -1;
+			if (endEntityIndex > ch.length) {

```

Comment:
Why you did that?

Perhaps you try to fix an another bug. If it that please rénové this code and create an another PR

Reply:
This was the fix for the exception that Fred brought up here: https://github.com/eclipse/lemminx/issues/1368#issuecomment-1313924572

Should I do that in another PR?

Reply:
Ok let's keep like this.